src/client/src/Stream.c

```diff
@@ -296,20 +306,32 @@ STATUS createStream(PKinesisVideoClient pKinesisVideoClient, PStreamInfo pStream
 
 CleanUp:
 
-    if (locked) {
+    if (clientLocked) {
         pKinesisVideoClient->clientCallbacks.unlockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoClient->base.lock);
     }
 
     if (STATUS_FAILED(retStatus) && tearDownOnError) {
+        if(!clientStreamsListLocked) {
+            pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoClient->base.streamListLock);
+            clientStreamsListLocked = TRUE;
+        }
         freeStream(pKinesisVideoStream);
     }
 
+    // Need to be under streamListLock while calling freeStream() so wait until after that check to unlock
+    if(clientStreamsListLocked) {
```

Comment:
spacing... Also, this didn't make sense. Perhaps revisit the logic of streamlist lock acquisition and release

Reply:
freeStream actually removed it from the client streams list, hence any call to it must happen under streamlist lock.

Reply:
I saw that you are wrapping freeStream which is OK. Perhaps I commented it earlier or something like it. Ignore it